lib/in_app_update.dart

```diff
@@ -78,6 +81,10 @@ class InAppUpdate {
     );
   }
 
+  static Stream<InstallStatus> get installUpdateListener {
+    return _installListener.receiveBroadcastStream().cast<InstallStatus>();
```

Comment:
@ViniciusSossela This cast isn't going to work, since the method channel cannot send enum values directly.
(they are not serializable by the BinaryMessenger, see https://docs.flutter.dev/platform-integration/platform-channels?tab=type-mappings-kotlin-tab#codec)

Instead, you'll have to map the integer value of Android's `com.google.android.play.core.install.model.InstallStatus` enum to the relevant values in the Dart enum.

Something like 

```
static Stream<InstallStatus> get installUpdateListener {
  return _installListener.receiveBroadcastStream().cast<int>().map((int value) {
    switch(value) {
      case 0:
        return InstallStatus.unknown;
      case 1:
        return InstallStatus.pending;
      case 2:
        return InstallStatus.downloading;
      case 3:
        return InstallStatus.installing;
      case 4:
        return InstallStatus.installed;
      case 5:
        return InstallStatus.failed;
      case 6:
        return InstallStatus.canceled;
      case 11:
        return InstallStatus.downloaded;
      default:
        return InstallStatus.unknown;
    }
  });
}
```

And maybe add a documentation link to https://developer.android.com/reference/com/google/android/play/core/install/model/InstallStatus#summary

Reply:
done